### ISSUE TITLE
found 8 vulnerabilities (6 low, 2 high)

### ISSUE THREAD

**Issue by adriatic** [gatsbyjs/gatsby#29048]
## Description

Running the simplest gatsby app ( created by cloning `/gatsby-starter-hello-world.git`results with `found 8 vulnerabilities (6 low, 2 high)`

***I disagree with the often stated opinion that one would not worry about vulnerabilities reported by npm.***

### Steps to reproduce

Clear steps describing how to reproduce the issue. **Please** link to a reproduction, this makes your issue _much_ easier to diagnose (seriously).

How to Make a Minimal Reproduction: https://www.gatsbyjs.com/contributing/how-to-make-a-reproducible-test-case/

Create the application using the command

```
gatsby new tutorial-part-two https://github.com/gatsbyjs/gatsby-starter-hello-world
```

### Expected result

What should happen?

A clean build, with no vulnerabilities in Gatsby dependencies

### Actual result

What happened.

```
 gatsby new tutorial-part-two https://github.com/gatsbyjs/gatsby-starter-hello-world
info Creating new site from git: https://github.com/gatsbyjs/gatsby-starter-hello-world.git

Cloning into 'tutorial-part-two'...
remote: Enumerating objects: 16, done.
remote: Counting objects: 100% (16/16), done.
remote: Compressing objects: 100% (11/11), done.
remote: Total 16 (delta 0), reused 9 (delta 0), pack-reused 0
success Created starter directory layout
info Installing packages...


> bufferutil@4.0.2 install C:\work\learning\gatsby\tutorial-part-two\node_modules\bufferutil
> node-gyp-build


> utf-8-validate@5.0.3 install C:\work\learning\gatsby\tutorial-part-two\node_modules\utf-8-validate
> node-gyp-build


> core-js@2.6.11 postinstall C:\work\learning\gatsby\tutorial-part-two\node_modules\babel-runtime\node_modules\core-js
> node -e "try{require('./postinstall')}catch(e){}"

Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!

The project needs your help! Please consider supporting of core-js on Open Collective or Patreon:
> https://opencollective.com/core-js
> https://www.patreon.com/zloirock

Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)


> core-js@3.7.0 postinstall C:\work\learning\gatsby\tutorial-part-two\node_modules\core-js
> node -e "try{require('./postinstall')}catch(e){}"


> core-js-pure@3.7.0 postinstall C:\work\learning\gatsby\tutorial-part-two\node_modules\core-js-pure
> node -e "try{require('./postinstall')}catch(e){}"


> gatsby-telemetry@1.4.1 postinstall C:\work\learning\gatsby\tutorial-part-two\node_modules\gatsby-telemetry
> node src/postinstall.js || true


> gatsby-cli@2.13.1 postinstall C:\work\learning\gatsby\tutorial-part-two\node_modules\gatsby\node_modules\gatsby-cli
> node scripts/postinstall.js


> gatsby@2.26.1 postinstall C:\work\learning\gatsby\tutorial-part-two\node_modules\gatsby
> node scripts/postinstall.js

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.3 (node_modules\chokidar\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

added 1886 packages from 1001 contributors and audited 1892 packages in 36.35s

134 packages are looking for funding
  run `npm fund` for details

found 8 vulnerabilities (6 low, 2 high)
```

### Environment

```

  System:
    OS: Windows 10 10.0.19041
    CPU: (8) x64 Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz
  Binaries:
    Node: 12.19.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.14.7 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 2.7.18 - /c/Python27/python
  Browsers:
    Chrome: 87.0.4280.141
    Edge: Spartan (44.19041.423.0), Chromium (87.0.664.75)
  npmPackages:
    gatsby: ^2.26.1 => 2.26.1 
  npmGlobalPackages:
    gatsby-cli: 2.17.1
```

**Comment by LekoArts** [comment]
Thank you for opening this!

Most if not all of those vulnerabilities don't affect Gatsby's generated production code as it all happens on the server, the clients only get HTML/CSS/JS served. We've merged https://github.com/gatsbyjs/gatsby/pull/28545 which will be released with 2.31 tomorrow. You then can update your project (and we'll also be updating the starters).

We're marking this issue as answered and closing it for now but please feel free to comment here if you would like to continue this discussion. We also recommend heading over to our [communities](https://www.gatsbyjs.org/contributing/community/#where-to-get-support) if you have questions that are not bug reports or feature requests. We hope we managed to help and thank you for using Gatsby!

**Comment by adriatic** [comment]
I re-discovered this (my completely forgotten) post searching about npm reported vulnerabilities today (using Gatsby CLI version: 3.6.0, which is a lot newer than the 2.31, that contains a significant number of upgrades to remove vulnerabilities and peerDependency warnings @LekoArts  mentions above. 

Today npm reports `found 38 vulnerabilities (34 moderate, 3 high, 1 critical)`  😄 

@LekoArts would you mind staring your today's position on npm reported vulnerabilities?
